tests/Agent/IntegrationTests/IntegrationTestHelpers/RemoteServiceFixtures/RemoteApplicationFixture.cs

```diff
@@ -225,6 +225,25 @@ private string FormatExitCode(int? exitCode)
             return exitCode.Value.ToString("X8");
         }
 
+        /// <summary>
+        /// Adds or replaces known problems to check for. This is used to check for things like transaction garbage collected.
+        /// Add an empty/null string[] with keepDefaults = false to clear the defaults.
+        ///
+        /// Includes by default:
+        /// - AgentLogBase.TransactionEndedByGCFinalizerLogLineRegEx
+        /// </summary>
+        /// <param name="problems">Regex values to check for from AgentLogBase.</param>
+        /// <param name="keepDefaults">If true, the default problems will be kept. If false, the default problems will be cleared.</param>
+        public void KnownProblemsToCheck(bool keepDefaults = true, params string[] problems)
```

Comment:
You're just including this in case we need it later, right? Maybe a name like `SetKnownProblems` or `OverrideKnownProblems` would be more descriptive?

Reply:
Yeah, those are better - naming is hard - next PR